src/main/kotlin/io/quarkus/code/ShortUrlResource.kt

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.code
+
+import io.quarkus.code.model.QuarkusProject
+import io.quarkus.code.model.ShortUrl
+import org.eclipse.microprofile.openapi.annotations.Operation
+import java.net.URI
+import javax.inject.Inject
+import javax.validation.Valid
+import javax.ws.rs.*
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+@Path("/s")
+class ShortUrlResource {
+    @Inject
+    lateinit var urlRepository: UrlRepository
+
+    @GET
+    @Path("/")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Operation(summary = "Create a short url based on the parameters")
+    fun createShort(@Valid @BeanParam project: QuarkusProject): Response {
+        val url = "https://code.quarkus.io/api/download?g=${project.groupId}&a=${project.artifactId}&v=${project.version}&c=${project.className}&e=${project.extensions}"
+        val response = { shortUrl: ShortUrl ->
+            Response.ok("https://code.quarkus.io/s/${shortUrl.id}").build()
+        }
+        urlRepository.getByUrl(url)?.let { shortUrl ->
```

Comment:
You may get some race condition here (is it still safe?)

Reply:
I don't see how this would be a problem, it's checking if there is already a url for these parameters. If 2 of them exist in the db it's also not a problem

Reply:
Technically we could also always create a url regardless, it's just a small optimization

Reply:
No problem, that was just to be sure you had it thought.